python/error.hpp

```diff
@@ -18,18 +17,16 @@ namespace pybind11 { namespace detail {
 
 namespace pyarb {
 
-using arb::arbor_exception;
-
 // Python wrapper errors
 
-struct python_error: arbor_exception {
-    explicit python_error(const std::string& message);
+struct pyarb_error: std::runtime_error {
+    pyarb_error(const std::string& what_msg):
+        std::runtime_error(what_msg) {}
 };
 
-template <typename T, typename F>
-T&& assert_predicate(T&& t, F&& f, const char* msg) {
-    if (!f(t)) throw std::runtime_error(msg);
-    return std::forward<T>(t);
+static
```

Comment:
Simplify this by not passing the predicate, because the test does not need to be performed in the function.
The predicate is passed to the py2optional function, because it needs to be applied after the cast.